src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py

```diff
@@ -1146,6 +1146,74 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group, vn
                  .format(database_engine, resource_group, replicas[1]), checks=NoneCheck())
 
 
+class FlexibleServerVirtualEndpointSwitchoverMgmtScenarioTest(ScenarioTest):  # pylint: disable=too-few-public-methods
+
+    postgres_location = 'eastus'
+
+    @AllowLargeResponse()
+    @ResourceGroupPreparer(location=postgres_location)
+    def test_flexible_server_virtual_endpoint_mgmt(self, resource_group):
+        self._test_flexible_server_virtual_endpoint_mgmt('postgres', resource_group)
+
+    def _test_flexible_server_virtual_endpoint_mgmt(self, database_engine, resource_group):
+        location = self.postgres_location
+        primary_role = 'Primary'
+        replica_role = 'AsyncReplica'
+        master_server = self.create_random_name(SERVER_NAME_PREFIX, 32)
+        replica = self.create_random_name(F'azuredbclirep', SERVER_NAME_MAX_LENGTH)
+        master_vnet_args = '--public-access none'
+        virtual_endpoint_name = self.create_random_name(F'virtual-endpoint', 32)
+        read_write_endpoint_type = 'ReadWrite'
+
+        # create a server
+        self.cmd('{} flexible-server create -g {} --name {} -l {} --storage-size {} {} --tier GeneralPurpose --sku-name Standard_D2s_v3 --yes'
+                 .format(database_engine, resource_group, master_server, location, 256, master_vnet_args))
+        primary_result = self.cmd('{} flexible-server show -g {} --name {} '
+                          .format(database_engine, resource_group, master_server),
+                          checks=[JMESPathCheck('replicationRole', primary_role)]).get_output_in_json()
```

Comment:
since we have added new replica property bag, I think it would be better if we start validating against latest set of properties. That way we can get some coverage for new properties as well